sockets/challonge.js

```diff
@@ -0,0 +1,107 @@
+var config = require('../config');
+var XMLHttpRequest = require('xmlhttprequest').XMLHttpRequest;
+module.exports = function(io) {
+    var challongeData = {};
+    var challongePollFrequency = 10000;
+    var connectedSockets = 0;
+    var challongeApiRoot = 'https://api.challonge.com/v1';
+    var challongeHash = null;
+    var matches = [];
+    var players = [];
+
+    var challongeIO = io.of('/challonge');
+
+    challongeIO.on('connection', function(socket) {
+        console.log('Challonge connected');
+        socket.emit('update challonge', challongeData);
+        connectedSockets++;
+
+        socket.on('disconnect', function() {
+            console.log('Challonge disconnected');
+            connectedSockets--;
+        });
+
+        socket.on('update challonge', function(msg) {
+            challongeData = msg;
+            if (challongeData.challongeUrl) {
+                if (!challongeHash) {
+                    challongeHash = createChallongeHash(challongeData.challongeUrl);
+                }
+
+                getchallongePollableData(challongeData);
+                setTimeout(pollChallonge, challongePollFrequency);
+            }
+
+            challongeIO.emit('update challonge', challongeData);
+            console.log('update challonge: ' + JSON.stringify(challongeData));
```

Comment:
Does this need to be here? It looks like `getchallongePollableData` also calls emit.